Loop/Managers/DeviceDataManager.swift

```diff
@@ -1329,12 +1329,18 @@ extension DeviceDataManager {
             log.error("Remote Notification: Overrides not enabled.")
             return
         }
+        
+        let defaultServiceIdentifier = "NightscoutService"
+        let serviceIdentifer = notification["serviceIdentifier"] as? String ?? defaultServiceIdentifier
 
```

Comment:
A side point as I was working through this handlePushNotification code...

I'm thinking an evolution of this code, someday, may be to send a raw notification to each service for handling. They could handle all the validation, like this expiration check below, which may vary by service.

There are some remote actions maybe only they can handle like activating a service setting or mode.

On that matter, maybe they could handle all notifications (i.e. deliver the bolus/carbs/etc). We could allow services to perform such via service delegation and add controls on the Loop side to restrict things for safety. 

Anyways, just a few thoughts.

Reply:
I think validating the command is the service's responsibility. Not sure executing the command is, though. I think the "command set" is a Loop domain. Maybe we're talking about the same thing, though.